multiselect/add-new-item/MultiSelectAndAddButton/MultiSelectAndAddButton/Shared/MultiSelectWithAdd.razor

```diff
@@ -87,24 +112,15 @@
         // in a real case you may want to actually use the current user input and method, see more at
         // https://docs.telerik.com/blazor-ui/components/multiselect/events#onread
         var currOpts = await _dataService.GetOptions();
-        Options = new ObservableCollection<OptionsModel>(currOpts);
+        args.Data = Options = new List<OptionsModel>(currOpts);
 
-        // START of the hack for updating the values from an extrenal place
-        // generally, you should not update the Value here, or any other component parameter
-        // we do this here to make the component re-render after it has the new data, otherwise
-        // it could not match the new value we just added to the data source, and would not create a "tag"
-        if (shouldUpdateValueParam)
+        // hack for clearing the search textbox and raise another flag for OnAfterRenderAsync
+        if (shouldClearInput)
         {
-            shouldUpdateValueParam = false;
-
-            // to make the framework re-render a component, change the reference to the complex object
-            // see more at https://docs.telerik.com/blazor-ui/common-features/observable-data#refresh-data
-            List<int> updatedValues = new List<int>(Tags);
-            Tags = updatedValues;
+            shouldClearInput = false;
+            shouldUpdateTags = true;
 
-            // clear the input value now that the user has added the current text to the data source
-            // the sample JS method is in the index page of the blazor app, move it as necessary
             await _js.InvokeVoidAsync("clearInput", Id);
```

Comment:
Can we clear the input from code: https://docs.telerik.com/blazor-ui/components/grid/filter/searchbox#filter-from-code

Reply:
I don't understand how the Grid article is related to the MultiSelect. Am I missing something?

Reply:
Me too. I'm not sure what I was thinking. Disregard.